backend/bcssm_backend/decorators.py

```
@@ -0,0 +1,53 @@
+            logger.warning(
+                "Authentication error in %s: %s", f.__name__, e.message
+            )
+            return jsonify({'error': e.message}), 401
+        except NotFoundError as e:
+            logger.warning("Not found in %s: %s", f.__name__, e.message)
+            return jsonify({'error': e.message}), 404
+        except (DatabaseError, SQLAlchemyError) as e:
+            msg = e.message if isinstance(e, DatabaseError) else str(e)
+            logger.error("Database error in %s: %s", f.__name__, msg)
+            return jsonify({'error': 'Internal server error'}), 500
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)